msamp/fsdp/_runtime_utils.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""MS-AMP fsdp._runtime_utils module."""
+
+from typing import no_type_check
+
+import torch
+from torch.distributed.fsdp import FullyShardedDataParallel
+
+old_post_backward_hook = torch.distributed.fsdp._runtime_utils._post_backward_hook
+
+
+@no_type_check
+@torch.no_grad()
+def _fp8_post_backward_hook(state, handle, *unused):
+    """A post-backward communication hook which supports fp8."""
+    if not isinstance(state, FullyShardedDataParallel):
```

Comment:
Why is the old_post_backward_hook called when the state is not FullyShardedDataParallel?

Reply:
Good question. Seems state is always instance of FullyShardedDataParallel, so I remove the check `if not isinstance(state, FullyShardedDataParallel)`.